src/azul/plugins/metadata/hca/transform.py

```diff
@@ -82,6 +86,90 @@
 pass_thru_uuid4: PassThrough[api.UUID4] = PassThrough()
 
 
+class OrganismAge(FieldType[JSON, str]):
+
+    def to_index(self, value: JSON) -> str:
+        """
+        >>> OrganismAge().to_index({'value': '20', 'unit': 'year'})
+        '20 year'
+
+        >>> OrganismAge().to_index({'notval': '20', 'notu': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A dictionary with entries for `value` and `unit` is required
+
+        >>> OrganismAge().to_index({'value': '', 'unit': 'year'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not be empty
+
+        >>> OrganismAge().to_index({'value': '20', 'unit': ''})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not be empty
+
+        >>> OrganismAge().to_index({'value': None, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: A `unit` entry must be accompanied by a `value` entry
+
+        >>> OrganismAge().to_index({'value': '20 ', 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must not contain space characters
+
+        >>> OrganismAge().to_index({'value': 20, 'unit': 'years'})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `value` entry must be of string type
+
+        >>> OrganismAge().to_index({'value': '20', 'unit': 'years '})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must not contain space characters
+
+        >>> OrganismAge().to_index({'value': '20', 'unit': True})
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: The `unit` entry must be of string type
+
+        """
+        if value is None:
```

Comment:
The argument is declared `JSON` which precludes `None`. Remind me again, under what circumstances can the argument be `None`?

Reply:
It gets used in [elasticsearch_service.py](https://github.com/DataBiosphere/azul/blob/34cc7167cc378e210d5c718699610756c64ba6aa/src/azul/service/elasticsearch_service.py#L152)

Reply:
I changed the signature to account for the usage of `None`.

Reply:
If that's the only call site passing `None` we should change the call site and eliminate that special case since it can also be achieved by passing a dict with `unit` and `value` set to `None`.

Reply:
Ah that makes sense, except the call site is for all field types, `organismAge` is the only special one with a measured term entry requiring a dict.

Reply:
See if you can avoid the ambiguity between `None` and `{"unit": None, "value": None}`. It is a bad smell.